testserv.c

```diff
@@ -87,12 +86,43 @@ mustdiallocal(int port)
     return fd;
 }
 
-#define SERVER() (progname = __func__, mustforksrv())
+#ifdef WITHCOV
+void __gcov_flush (void);
+
+void
+gcov_flush(int signum)
+{
+    __gcov_flush(); /* dump coverage data  */
```

Comment:
No need to call `__gcov_flush` here. Instead we can call `exit` and let gcov's atexit handler flush.

It's better not to call gcov internals directly if we can avoid it, and in this case we can.

Reply:
Done